Automattic/wp-calypso
# Marketplace: premium plugins now require a personal plan.

#### Proposed Changes

* Introduces a new `marketplace-personal-premium` feature flag so we can test the feature in wpcalypso environment
* Changes the required plan to purchase paid plugins from the WP.com marketplace from Business to Personal (if the feature flag above is enabled). 
* Removes `Pro` plan requirements as users on Starter already have this feature enabled and if they are on free, then the only path is Personal
* Checkout: respects 0 as a possible variant index.

Plugin Details Before| Plugin Details After
--- | ---
<img width="1116" alt="CleanShot 2022-08-01 at 21 00 39@2x" src="https://user-images.githubusercontent.com/12430020/182212533-29166446-81fe-4760-a459-0ed98f757d24.png">|<img width="1101" alt="CleanShot 2022-08-01 at 20 38 28@2x" src="https://user-images.githubusercontent.com/12430020/182209220-05f510ca-532b-4e69-ae99-8cb908bff45d.png">

Checkout Before | Checkout After
--- | ---
![CleanShot 2022-08-02 at 12 56 25@2x](https://user-images.githubusercontent.com/12430020/182347702-82e3fd88-5658-4c26-bad8-6309e0dcb654.png)|<img width="592" alt="CleanShot 2022-08-02 at 12 55 49@2x" src="https://user-images.githubusercontent.com/12430020/182347733-a83cfee2-16f1-4432-9db7-f61cd02ec9fe.png">


#### Testing Instructions

* Activate the feature flag by adding `?flags=marketplace-personal-premium` to the URL.
* Switch to a Simple sites with a Free plan.
* Go to Plugins.
* Select a paid plugin.
* Make sure that info about the Starter plan is displayed below the "Purchase and activate" button.
* Click on the "Purchase and activate" button and proceed with the checkout process.
* Make sure you get the Personal plan. 

Fixes https://github.com/Automattic/wp-calypso/issues/65523
